core/lib/multivm/src/versions/vm_fast/vm.rs

```diff
@@ -375,11 +388,18 @@ impl<S: ReadStorage> Vm<S> {
     }
 
     pub(crate) fn decommitted_hashes(&self) -> impl Iterator<Item = U256> + '_ {
-        self.inner.world_diff.decommitted_hashes()
+        self.inner.world_diff().decommitted_hashes()
     }
 
-    pub(super) fn gas_remaining(&self) -> u32 {
-        self.inner.state.current_frame.gas
+    pub(super) fn gas_remaining(&mut self) -> u32 {
```

Comment:
Ah, I've wanted to ask about this method 🙃 To understand better: I *think* this replicates the previous implementation (getting gas from the most recent `Callframe`, i.e., skipping near calls). So was the previous implementation incorrect?

Reply:
The callframe always contains the current gas, even if the execution is in a near call. It is designed that way because otherwise we'd have to check if we are in a near call all the time.